scripts/tests/upgrade-test.sh

```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+# Runs a test to determine the latency of certain user actions
+
+set -euo pipefail
+export RUST_LOG="${RUST_LOG:-info}"
+
+source scripts/_common.sh
+build_workspace
+add_target_dir_to_path
+
+devimint upgrade-test "$@"
```

Comment:
Perhaps we can improve the usability of this script if we use `nix_build_binary_for_version` from `scripts/_common.sh` and pass versions to test as arguments.

```bash
versions=("$@")
for fed_version in "${versions[@]}"; do
  echo "Running upgrade test from fedimintd $fed_version to current"
  fedimintd_path="$(nix_build_binary_for_version 'fedimintd' "$fed_version")"
  export FM_BACKWARDS_COMPATIBILITY_TEST=1
  devimint upgrade-test "$fedimintd_path"
done
```

Then we can call with `./scripts/tests/upgrade-test.sh v0.2.1 v0.2.2` (and also add a `just` command).

I needed to add FM_BACKWARDS_COMPATIBILITY_TEST to handle the long latency for upgrading v0.2.1.